src/math/trig.c

```diff
@@ -94,9 +108,159 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
+		 * angles greater than or equal to pi/4, we initialize (x,y)
+		 * along the y-axis. This improves the accuracy of the algorithm
+		 * near the edge of the domain of convergence
+		 */
+		if (b_sincosvalue > cord_arcsincos_q28fl) {
+			x = 0;
+			y = cord_arcsincos_q30fl;
+			z = 843314856;
+		} else {
+			x = cord_arcsincos_q30fl;
+			y = 0;
+			z = 0;
+		}
+	}
+	if (type == EN_32B_CORDIC_ACOSINE) {
+		/* Initialize the variables for the cordic iteration
+		 * angles less than pi/4, we initialize (x,y) along the x-axis.
+		 * angles greater than or equal to pi/4, we initialize (x,y)
+		 * along the y-axis. This improves the accuracy of the algorithm
+		 * near the edge of the domain of convergence
+		 */
+		if (b_sincosvalue < cord_arcsincos_q28fl) {
+			x = 0;
+			y = cord_arcsincos_q30fl;
+			z = 843314856;
+		} else {
+			x = cord_arcsincos_q30fl;
+			y = 0;
+			z = 0;
+		}
+	}
+	b_sincosvalue <<= 1U;
+	/* DCORDIC(Double CORDIC) algorithm */
+	/* Double iterations method consists in the fact that unlike the classical */
+	/* CORDIC method,where the iteration step value changes EVERY time, i.e. on */
+	/* each iteration, in the double iteration method, the iteration step value */
+	/* is repeated twice and changes only through one iteration */
+	i = numiters - 1;
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (int16_t)((b_i + 1) << 1U);
+		if (j >= 31)
+			j = 31;
 
+		if (b_i < 31)
+			k = (int16_t)b_i;
+		else
+			k = 31;
+
+		xshift = (x >>	(k));
+		xdshift = (x >> (j));
+		yshift = (y >>	(k));
+		ydshift = (y >> (j));
+
+		if (type == EN_32B_CORDIC_ASINE) {
+			/* Do nothing if x currently equals the target value. Allowed for
+			 * double rotations algorithms, as it is equivalent to rotating by
+			 * the same angle in opposite directions sequentially. Accounts for
+			 * the scaling effect of CORDIC Pseudo-rotations as well.
+			 */
+			if (y == b_sincosvalue) {
+				x += xdshift;
+				y += ydshift;
+			} else if (((y >= b_sincosvalue) && (x >= 0)) ||
+				  ((y < b_sincosvalue) && (x < 0))) {
+				x = (x - xdshift) + yshift;
+				y = (y - ydshift) - xshift;
+				z -= LUT[b_i];
+			} else {
+				x = (x - xdshift) - yshift;
+				y = (y - ydshift) + xshift;
+				z += LUT[b_i];
+			}
+		}
+		if (type == EN_32B_CORDIC_ACOSINE) {
+			/* Do nothing if x currently equals the target value. Allowed for
+			 * double rotations algorithms, as it is equivalent to rotating by
+			 * the same angle in opposite directions sequentially. Accounts for
+			 * the scaling effect of CORDIC Pseudo-rotations as well.
+			 */
+			if (x == b_sincosvalue) {
+				x += xdshift;
+				y += ydshift;
+			} else if (((x > b_sincosvalue) && (y >= 0)) ||
+				  ((x < b_sincosvalue) && (y < 0))) {
+				x = (x - xdshift) - yshift;
+				y = (y - ydshift) + xshift;
+				z += LUT[b_i];
+			} else {
+				x = (x - xdshift) + yshift;
+				y = (y - ydshift) - xshift;
+				z -= LUT[b_i];
+			}
+		}
+
+		b_sincosvalue += (b_sincosvalue >> ((uint32_t)j));
```

Comment:
I'd make `j` uint32_t type to avoid typecast and ensure it doesn't underflow. Remove the extra ()s too.